### PR TITLE
Add document_type only query to email-alert-api

### PR DESCRIPTION
**Pull request by tuzz**
https://trello.com/c/gPjk5P2A/611-add-document-type-only-query-to-email-alert-api

### CODE REVIEW

File: spec/workers/notification_worker_spec.rb [hunk 1]
```diff
@@ -102,6 +102,24 @@ def make_it_perform
       end
     end
 
+    context "when the subscriber list has no tags or links" do
+      before do
+        create(:subscriber_list, document_type: 'travel-advice', tags: {}, links: {})
+      end
+
+      it "sends a bulletin when document_type matches" do
```
Comment (bradwright) [inline spec/workers/notification_worker_spec.rb]: Could we add a test for when the bulletin has some tags as well?

Comment (bradwright) [inline spec/workers/notification_worker_spec.rb]: Also the case where it should trigger more than 1 list (eg "all countries" + "travel advice for spain")

**Comment by tuzz** [comment]
Addressed @bradwright – please could you take another look.

File: spec/models/subscriber_list_spec.rb [hunk 2]
```diff
@@ -29,6 +29,46 @@
     end
   end
 
+  describe "validations" do
+    subject { FactoryGirl.build(:subscriber_list) }
```
Comment (tijmenb) [inline spec/models/subscriber_list_spec.rb]: Would the tests be more independently readable if `subject` were changed to `let(:subscriber_list)`?

Comment (tuzz) [inline spec/models/subscriber_list_spec.rb]: Can do. I'm somewhat indifferent. `subject` is supposed to be the main object that you're testing (system under test).

**Comment by bradwright** [comment]
Merging on the basis that this is feature complete and tested, and I want to turn some things off.

**Pull request merged**